### DISCUSSION TITLE
Ubuntu Python 3.10, setuptools, and PEP 440 parse errors?

### DISCUSSION THREAD

**Discussion by Tbruno25** [python-poetry/poetry#6623] [General]
@neersighted Thank you for the thoughtful response to my question within issue #6013

I was trying to wrap my head around the prevalence of PEP 440 errors running Ubuntu and would like to share my findings -- 

```
$ lsb_release -a
Distributor ID:	Linuxmint
Description:	Linux Mint 21
Release:	21
Codename:	vanessa

$  python --version 
Python 3.10.6
```

If I remove poetry, re-install, and try to `poetry install` I get the following PEP 440 error:
```
$  curl -sSL https://install.python-poetry.org | python3 - --uninstall
$  rm -rf ~/.cache/pypoetry
$  rm -rf ~/.local/share/virtualenv/py_info
$  curl -sSL https://install.python-poetry.org | python3 -
Installing Poetry (1.2.1): Done

Poetry (1.2.1) is installed now. Great!

$  cd dev_env
$  poetry install
Creating virtualenv dev-env in /home/projects/dev_env/.venv

Invalid PEP 440 version: '0.1.43ubuntu1'
```

I came across [this discussion](https://github.com/pypa/setuptools/issues/3278) regarding setuptools and python3.10 so I tried downgrading setuptools
```
$  pip show setuptools
Name: setuptools
Version: 65.4.0

$ pip install setuptools==59.8.0
Successfully installed setuptools-59.8.0

$ rm -r .venv
$ poetry install 
Installing the current project: dev_env (0.1.2)  # success!
```

Poetry still functions if I upgrade setuptools back to the latest version
```
$ pip install setuptools --upgrade
Successfully installed setuptools-65.4.0

$  poetry install
Installing the current project: dev_env (0.1.2)
```

But if I delete the cache I get the original PEP 440 error
```
$ rm -rf ~/.local/share/virtualenv/py_info
$ poetry install
Invalid PEP 440 version: '0.1.43ubuntu1'
```

This obviously seems more like a bug with python rather poetry but I wanted to share these findings as it can directly lead to a poor user experience with poetry!

**Comment by neersighted** [comment]
I'm not sure there's enough info here to figure out what might be going on -- however, there is enough for someone who is sufficiently motivated to dig in. If this truly is related to setuptools/distutils and Python 3.10's interaction with Debian patching, this might be resolved on its own. If not, hopefully someone else who wants to do some root cause analysis has time to take a look at it.

**Comment by dargueta** [comment]
It's a bug (ish) in Ubuntu: https://bugs.launchpad.net/ubuntu/+source/python-debian/+bug/1926870

If you're feeling brave and have sudo access, you can fix it manually.

1. Go to `/usr/lib/python3/dist-packages/`
2. Find the directory named `python_debian-0.1.43ubuntu1.egg-info` and rename it to `python_debian-0.1.43+ubuntu1.egg-info` (note the plus sign).
3. In this directory, open the file named PKG-INFO (requires sudo)
4. Edit the version number to match the name of the directory, like so:

![image](https://user-images.githubusercontent.com/620513/193427903-75e599b6-973d-4a28-bb5b-62dc4372012e.png)

5. Save
6. Pray to your favorite deity and run `poetry install` again

Fortunately you'll only need to do this once.

**Reply by Tbruno25** [reply]
@dargueta Solved it for me :+1: Thanks so much for the fix!

**Reply by dargueta** [reply]
Great! Glad to be of service.
